### PR TITLE
Add Aptos APT Price

### PR DESCRIPTION
**Pull request by chef-seaweed**
### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: models/prices/aptos/prices_aptos_tokens.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{{ config(
+        schema='prices_aptos',
+        alias = alias('tokens'),
```
Comment (Hosuke) [inline models/prices/aptos/prices_aptos_tokens.sql:3]:
```suggestion
        alias = 'tokens',
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/prices/aptos/prices_aptos_tokens.sql (hunk 1 above)
Comment (chef-seaweed) [reply models/prices/aptos/prices_aptos_tokens.sql]: My bad. Thank you @Hosuke for the help 🙏

Comment (chef-seaweed) [reply models/prices/aptos/prices_aptos_tokens.sql]: For APT contract address, should I change
`cast(contract_address as varchar) as contract_address` to `from_hex(contract_address) as contract_address

File: models/prices/aptos/prices_aptos_tokens.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        materialized='table',
+        file_format = 'delta',
+        tags = ['static', 'dunesql']
+        )
+}}
+SELECT 
+    token_id
+    , blockchain
+    , symbol
+    , cast(contract_address as varchar) as contract_address
```
Comment (Hosuke) [inline models/prices/aptos/prices_aptos_tokens.sql:13]:
```suggestion
    , from_base58(contract_address) as contract_address
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/prices/aptos/prices_aptos_tokens.sql (hunk 1 above)
Comment (Hosuke) [reply models/prices/aptos/prices_aptos_tokens.sql]: It's not hexstring, let's try `from_base58` first.

File: models/prices/aptos/prices_aptos_tokens.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        materialized='table',
+        file_format = 'delta',
+        tags = ['static', 'dunesql']
+        )
+}}
+SELECT 
+    token_id
+    , blockchain
+    , symbol
+    , from_base58(contract_address) as contract_address
```
Comment (Hosuke) [inline models/prices/aptos/prices_aptos_tokens.sql:13]: Hi @chef-seaweed, we need to make sure this column is `varbinary` type.
Do you have any suggestion or idea?

(In solana spell we use `from_base58` to handle this column, but seems it does not work at here.)

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/prices/aptos/prices_aptos_tokens.sql (hunk 3 above)
Comment (chef-seaweed) [reply models/prices/aptos/prices_aptos_tokens.sql:13]: Yeah  seems like we don't have a proper function to convert Aptos token address to `varbinary` type yet. 
Also we'll have `Error: Not a hexadecimal digit: ":"` with `from_hex`
Should we build a dedicated price table for Aptos?

Comment (Hosuke) [reply models/prices/aptos/prices_aptos_tokens.sql:13]: Let me think if we need a new udf here.

File: models/prices/aptos/prices_aptos_tokens.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    , decimals
+FROM
+(
+    VALUES
+    ('apt-aptos','aptos','APT','0x1::aptos_coin::AptosCoin',8)
```
Comment (Hosuke) [inline models/prices/aptos/prices_aptos_tokens.sql:18]: Do you have any official explorer link for this token?

Comment (chef-seaweed) [reply models/prices/aptos/prices_aptos_tokens.sql:18]: https://aptoscan.com/coin/0x1::aptos_coin::AptosCoin

**Comment by jeff-dude** [comment]
thanks both. let me get this passed internally to see the plan for aptos in spellbook

**Comment by jeff-dude** [comment]
@chef-seaweed we aren't quite ready yet to add aptos into spellbook fully. for now, can you simply add the aptos token to `prices_native_tokens.sql`?

the native token spell doesn't require an address, so we can simply add the coinpaprika ID and symbol to obtain pricing info on the main token.

**Comment by chef-seaweed** [comment]
> @chef-seaweed we aren't quite ready yet to add aptos into spellbook fully. for now, can you simply add the aptos token to `prices_native_tokens.sql`?
> 
> the native token spell doesn't require an address, so we can simply add the coinpaprika ID and symbol to obtain pricing info on the main token.
Screenshot 2024-07-25 at 4 44 12 PM


Got it, but I believe `APT` is already in `prices_native_tokens.sql` so should we close this PR.

**Comment by jeff-dude** [comment]
yeah, lets close for now. more to come!

**Pull request closed without merging**